src/main/java/zean/Parser.java

```diff
@@ -0,0 +1,176 @@
+package zean;
+
+import java.util.ArrayList;
+import java.util.regex.Pattern;
+
+import zean.exception.DukeException;
+import zean.task.Deadline;
+import zean.task.Event;
+import zean.task.Task;
+import zean.task.Todo;
+
+/**
+ * The class that provides the parser of the input.
+ *
+ * @author Zhong Han
+ */
+public class Parser {
+
+    private static final Pattern DATE_PATTERN = Pattern.compile("\\d\\d\\d\\d-\\d\\d-\\d\\d");
+
+    /**
+     * Parses the input and calls the corresponding methods.
+     *
+     * @param input The input from the user.
+     * @param tasks The taskList object to store the tasks.
+     * @return The output to be printed on the console.
+     * @throws DukeException An exception related to the chatbot.
+     */
+    public static String parse(String input, TaskList tasks) throws DukeException {
+        if (input.isBlank()) {
+            throw new DukeException("OOPS!!! You forgot to type in anything!");
```

Comment:
I like how you provided different error messages for handling different possible cases when the user provides invalid inputs 👍